tools/local_provisioning_profile_finder/local_provisioning_profile_finder.py

```diff
@@ -0,0 +1,67 @@
+import argparse
+import datetime
+import plistlib
+import shutil
+import subprocess
+from typing import List, Optional, Tuple
+
+
+def _build_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("name", help="The name of the profile to find")
+    parser.add_argument("output", help="The path to copy the profile to")
+    parser.add_argument(
+        "--local-profiles",
+        nargs="*",
+        required=True,
+        help="All local provisioning profiles to search through",
+    )
+    parser.add_argument(
+        "--fallback-profiles",
+        nargs="+",
+        required=True,
+        help="Fallback provisioning profiles to use if not found locally",
+    )
+    return parser
+
+
+def _profile_contents(profile: str) -> Tuple[str, datetime.datetime]:
+    output = subprocess.check_output(["security", "cms", "-D", "-i", profile])
+    plist = plistlib.loads(output)
+    return plist["Name"], plist["CreationDate"]
+
+
+def _find_newest_profile(name: str, profiles: List[str]) -> Optional[str]:
+    sanitized_name = name.replace("_", " ")
+    newest_path: Optional[str] = None
+    newest_date: Optional[datetime.datetime] = None
+    for profile in profiles:
+        name, creation_date = _profile_contents(profile)
+        if name != sanitized_name:
```

Comment:
Rare but it might be worth adding `OR team_id != team_id`

and by rare I mean of course this applies to us

Reply:
(Unless you can extract the bundle ID like you mentioned above which I do see in the Finder preview but not in the output of the security tool)

Reply:
Adding this as we also had this case